routes/api/v1/meals.js

```diff
@@ -0,0 +1,51 @@
+var express = require('express');
+var router = express.Router();
+var Meal = require('../../../models').Meal;
+var Food = require('../../../models').Food;
+var MealFoods = require('../../../models').MealFoods;
+
+
+const parseMeals = (meals) => {
+   return meals.map(getMealObjects)
+}
+
+//The async/await here for parseFoods turned out to be key to getting the array of foods
+var getMealObjects = async (meal) => {
+  var getMealComponents = { id: meal.id,
+                          name: meal.name,
+                         foods: await parseFoods(meal.getFoods())
+                         }
+  return getMealComponents
+}
+
+var parseFoods = (foods) => {
+  return foodsArray = foods.map(getAllFoods)
```

Comment:
Nice work on all of these helper methods, great solve to get everything structured correctly.

Reply:
Hopefully you can re-use parseFoods and getAllFoods in the GET meals by id, but I'm not sure exactly what that would look like offhand.